ballerina/main.bal

```diff
@@ -141,8 +171,57 @@ isolated function handleParseResponseError(error chatResponseError) returns erro
     return chatResponseError;
 }
 
-isolated function getExpectedResponseSchema(typedesc<anydata> expectedResponseTypedesc) returns map<json> {
+// isolated function getExpectedResponseSchema(typedesc<anydata> expectedResponseTypedesc) returns map<json> {
+//     // Restricted at compile-time for now.
+//     typedesc<json> td = checkpanic expectedResponseTypedesc.ensureType();
+//     return generateJsonSchemaForTypedescAsJson(td);
+// }
+
+isolated function getExpectedResponseSchema(typedesc<anydata> expectedResponseTypedesc) returns SchemaResponse {
     // Restricted at compile-time for now.
     typedesc<json> td = checkpanic expectedResponseTypedesc.ensureType();
-    return generateJsonSchemaForTypedescAsJson(td);
+    return generateJsonObjectSchema(generateJsonSchemaForTypedescAsJson(td));
+}
+
+isolated function generateJsonObjectSchema(map<json> schema) returns SchemaResponse {
+    string[] supportedMetaDataFields = ["$schema", "$id", "$anchor", "$comment", "title", "description"];
+
+    // Check if the schema already has type "object"
+    if schema.hasKey("type") {
+        json typeValue = schema["type"];
+        
+        // If type is already "object", return as-is
+        if typeValue is string && (typeValue.toString() == "object") {
+            return {schema};
+        }
+    }
+    
+    // Create the new wrapper object schema
+    map<json> updatedSchema = {};
+    
+    // Copy all existing schema metadata (except type)
+    foreach var [key, value] in schema.entries() {
+        if supportedMetaDataFields.indexOf(key) is int {
+            updatedSchema[key] = value;
+        }
+    }
+    
+    // Set the type to "object"
+    updatedSchema["type"] = "object";
+    
+    map<json> properties = {};
+    map<json> content = {};
+    
+    // Copy all original schema definitions into content
+    foreach var [key, value] in schema.entries() {
+        if supportedMetaDataFields.indexOf(key) is int {
+            continue; // Skip metadata fields
+        }
+        content[key] = value;
+    }
+    
+    properties[OBJECT_KEY] = content.cloneReadOnly();
```

Comment:
Similarly for properties, why are you initializing first and then updating the property, which then requires creating a copy?